pkgs/by-name/sh/shark/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  boost,
+  cmake,
+  fetchFromGitHub,
+  openssl,
+  stdenv,
+}:
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "shark";
+    version = "4.1.0";
+
+    src = fetchFromGitHub {
+      owner = "Shark-ML";
+      repo = "Shark";
+      rev = "16a7cecf1c012ceaa406e3a5af54d1a6a47d5cda";
+      hash = "sha256-xwniI2+Kry04zQqlYjMTp60O6YLibFy+Q/2CY0PHpqs=";
+    };
+
+    # https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/tree/develop/SuperBuild/patches/SHARK?ref_type=heads
+    # patch of hdf5 seems to be not needed based on latest master branch of shark as HDF5 has been removed
+    # c.f https://github.com/Shark-ML/Shark/commit/221c1f2e8abfffadbf3c5ef7cf324bc6dc9b4315
+        patches = [
+            ./shark-2-ext-num-literals-all.diff
+        ];
+
+    # https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/blob/develop/SuperBuild/CMake/External_shark.cmake?ref_type=heads
+    cmakeFlags = [
+      "-DBUILD_SHARED_LIBS=ON"
+      "-DBUILD_EXAMPLES=OFF"
+      "-DBUILD_DOCS=OFF"
+      "-DBUILD_TESTING=OFF"
+      "-DENABLE_CBLAS=OFF"
+      "-DENABLE_OPENMP=OFF" # otb has this as optional flag during superbuild, make this as optional flag in nix ?
+    ];
+    buildInputs = [
+      boost
+      openssl
+    ];
+
+    nativeBuildInputs = [cmake];
+
+    meta = {
+      description = "Shark is a fast, modular, general open-source C++ machine learning library";
+      homepage = "http://shark-ml.github.io/Shark/";
```

Comment:
```suggestion
      homepage = "https://shark-ml.github.io/Shark/";
```

Reply:
Thanks @SuperSandro2000. I have updated it.